### PR TITLE
feat(content-api): reduce proxy size

### PR DESCRIPTION
**Pull request by spirit-wiegmann**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
     "esModuleInterop": true,
     "outDir": "dist/lib",
     "declarationDir": "dist/types",
+    "skipLibCheck": true,
```
Comment (nico-mcalley) [inline tsconfig.json:16]: according to https://www.typescriptlang.org/tsconfig#skipLibCheck we would loose type safety here. Can we drop it?

File: tsconfig.proxy.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    "strict": true,
+    "sourceMap": true,
+    "declaration": true,
+    "allowSyntheticDefaultImports": true,
+    "experimentalDecorators": true,
+    "emitDecoratorMetadata": true,
+    "esModuleInterop": true,
+    "outDir": "./proxy/dist/lib",
+    "declarationDir": "./proxy/dist/types",
+    "skipLibCheck": true,
```
Comment (nico-mcalley) [inline tsconfig.proxy.json:16]: according to https://www.typescriptlang.org/tsconfig#skipLibCheck we would loose type safety here. Can we drop it?

File: package.json [hunk 3]
```diff
@@ -92,13 +96,18 @@
       "@commitlint/config-conventional"
     ]
   },
+  "workspaces": [
```
Comment (nico-mcalley) [inline package.json:99]: This is basically symlinking the "proxy" folder to the root projects node_modules, right?

File: proxy/package.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "fsxa-proxy-api",
+  "version": "10.20.1-alpha.41",
+  "description": "The JavaScript Content API Library a.k.a Content API is an interface handling data coming from the FirstSpirit CaaS and the Navigation Service.",
```
Comment (nico-mcalley) [inline proxy/package.json:4]: the description should contain sth. like: "This package represents the proxy version of the content api, which runs on client side."

File: proxy/package.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  "keywords": [],
```
Comment (nico-mcalley) [inline proxy/package.json:5]: can we omit empty keywords?

File: proxy/package.json [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  "main": "dist/fsxa-proxy-api.cjs.js",
+  "module": "dist/fsxa-proxy-api.es5.js",
+  "types": "dist/types",
+  "author": "Tobias Golbs <golbs@e-spirit.com>",
```
Comment (nico-mcalley) [inline proxy/package.json:9]: feel free to set the author to your name

File: rollup.config.ts [hunk 7]
```diff
@@ -4,8 +4,11 @@ import resolve from '@rollup/plugin-node-resolve'
 import babel from '@rollup/plugin-babel'
 import { terser } from 'rollup-plugin-terser'
 import ts from 'rollup-plugin-ts'
+// @ts-ignore
```
Comment (nico-mcalley) [inline rollup.config.ts:7]: why?

File: package.json [hunk 8]
```diff
@@ -25,9 +25,12 @@
...
   },
   "scripts": {
-    "prebuild": "rimraf dist",
-    "build": "tsc --module commonjs --project tsconfig.build.json && rollup -c rollup.config.ts",
-    "build:local": "rimraf fsxa-api-*.tgz && npm version prerelease --no-git-tag-version --preid alpha && npm run build && npm pack",
+    "prebuild": "rimraf dist && rimraf proxy/dist",
+    "build:remote": "tsc --module commonjs --project tsconfig.build.json",
+    "build:proxy": "tsc --module commonjs --project tsconfig.proxy.json",
+    "build": "tsc --module commonjs --project tsconfig.build.json && tsc --module commonjs --project tsconfig.proxy.json && npm run build:remote && npm run build:proxy && rollup -c rollup.config.ts",
+    "version:dev": "export NEW_VER=$(npm version prerelease --no-git-tag-version --preid alpha); npm version --workspace proxy ${NEW_VER}",
```
Comment (nico-mcalley) [inline package.json:32]: oh my :-D how to test it?

File: proxy/README.md [hunk 9]
```diff
@@ -0,0 +1,392 @@
+# JavaScript Content API Library · Dedicated Client Package
+
+The JavaScript Content API Library a.k.a. Content API is an interface handling data coming from the FirstSpirit
+[CaaS](https://docs.e-spirit.com/module/caas/CaaS_Product_Documentation_EN.html) and the
+Navigation Service. The data is processed and transformed so that it can be used in any JavaScript project.
+
+> **_SubPackage_**
+> This package holds the compiled client part of the Content API. It is intended to be shipped in client environments,
```
Comment (nico-mcalley) [inline proxy/README.md:8]: what do you mean by compiled? I see source code

File: proxy/README.md [hunk 10]
```diff
@@ -0,0 +1,392 @@
...
+> such as the users' browser. It is optimized to hold only necessary parts needed for execution in a client context.
+> The main functionality is to talk to a backend service, implementing
+> the [FSXARemoteApi](https://github.com/e-Spirit/javascript-content-api-library/blob/master/src/modules/FSXARemoteApi.ts)
```
Comment (nico-mcalley) [inline proxy/README.md:11]: did you mean the FSXApi Interface? Otherwise I don't get the meaning

File: rollup.config.ts [hunk 11]
```diff
@@ -24,10 +27,15 @@ export default [
       },
     ],
     plugins: [
+      visualizer({
```
Comment (nico-mcalley) [inline rollup.config.ts:30]: how to use it?

File: rollup.config.ts [hunk 12]
```diff
@@ -37,6 +45,37 @@ export default [
...
+      ts(),
+      json(),
+      resolve({ extensions }),
+      commonjs(),
+      optimizeLodashImports(),
+      babel({
+        extensions,
+        include: ['src/proxy.ts'],
+        babelHelpers: 'bundled',
+      }),
```
Comment (nico-mcalley) [inline rollup.config.ts:75]: no more polyfills! good job 👍

**Review by nico-mcalley — CHANGES_REQUESTED** [review]
there are some issues I'd like to have fixed before merging.

also: why does this even work? I'd had assumed there would be more code moved to /proxy workspace. Maybe you can introduce me to that, because I'd really like to understand the solution in detail.

File: package.json (hunk 8 above)
Comment (spirit-wiegmann) [reply package.json:32]: Run `npm run version:dev` and look into the package.json of the Root project  and the proxy workspace. They should have the same, changed version.

File: rollup.config.ts (hunk 12 above)
Comment (spirit-wiegmann) [reply rollup.config.ts:74]: Thanks :)

**Review by nico-mcalley — APPROVED** [review]

**Pull request merged**